cmd/kubeadm/app/cmd/token_test.go

```diff
@@ -44,3 +50,69 @@ func TestRunGenerateToken(t *testing.T) {
 		t.Errorf("RunGenerateToken's output did not match expected regex; wanted: [%s], got: [%s]", TokenExpectedRegex, output)
 	}
 }
+
+func TestRunCreateToken(t *testing.T) {
+	var buf bytes.Buffer
+	fakeClient := &fake.Clientset{}
+	fakeClient.AddReactor("get", "secrets", func(action core.Action) (handled bool, ret runtime.Object, err error) {
+		return true, nil, errors.NewNotFound(v1.Resource("secrets"), "foo")
+	})
+
+	testCases := []struct {
+		name          string
+		token         string
+		usages        []string
+		extraGroups   []string
+		expectedError bool
+	}{
+		{
+			name:          "valid: empty token",
+			token:         "",
+			usages:        []string{"signing", "authentication"},
+			extraGroups:   []string{"system:bootstrappers:foo"},
+			expectedError: false,
+		},
+		{
+			name:          "valid: non-empty token",
+			token:         "abcdef.1234567890123456",
+			usages:        []string{"signing", "authentication"},
+			extraGroups:   []string{"system:bootstrappers:foo"},
+			expectedError: false,
+		},
+		{
+			name:          "valid: no extraGroups",
+			token:         "abcdef.1234567890123456",
+			usages:        []string{"signing", "authentication"},
+			extraGroups:   []string{},
+			expectedError: false,
+		},
+		{
+			name:          "invalid: incorrect token",
+			token:         "123456.AABBCCDDEEFFGGHH",
+			usages:        []string{"signing", "authentication"},
+			extraGroups:   []string{},
+			expectedError: true,
+		},
+		{
+			name:          "invalid: incorrect extraGroups",
+			token:         "abcdef.1234567890123456",
+			usages:        []string{"signing", "authentication"},
+			extraGroups:   []string{"foo"},
+			expectedError: true,
+		},
+		{
+			name:          "invalid: specifying --groups when --usages doesn't include authentication",
+			token:         "abcdef.1234567890123456",
+			usages:        []string{"signing"},
+			extraGroups:   []string{"foo"},
+			expectedError: true,
+		},
+		// TODO: add case about invalid usages once the validation is added.
```

Comment:
This should be added after #53929 merged.

Reply:
I would suggest to wait until #53929 merged, get here negative usages and then merge this PR. Otherwise looks ok for me.

Reply:
Agreed :)

Reply:
@kad updated.